Stb (#3823)
mlpack/mlpack: doc/user/load_save.md

mlpack::data::Save(outImages, matrix, info);
```

> ### Resize Image

Reviewer: ```suggestion
### Resize images
```
(to match the other subsection titles)

It is possible to resize images in mlpack with the following functions

 - `Resize(image, info, newWidth, newHeight)`

>    * `image` an armadillo matrix contains one column representing the image as
>    a flattened vector.
> 
>    * `info` will be populated with information from the image in `image`.
> 
>    * `newWidth` the new width of the resized image.
> 
>    * `newHeight` the new height of the resized image.
> 
>    * This function does not return but it modifies the info object and image
>      matrix.

Reviewer: If you think it is more effective, you could make a table of parameters like we do for other methods.  But, there are only four parameters, so, up to you.  If you want to keep this format I would suggest these changes:

```suggestion
   * Resize the image `image` to have width `newWidth` and height `newHeight`.
   * `image` should be an `arma::mat&` or similar matrix type with a single column representing the image to resize.
   * `info` should be a [`data::ImageInfo&`](#imageinfo) with the current size and parameters of `image`.
   * An exception will be thrown if the parameters of `info` do not match `image`.
   * `image` and `info` will be modified in-place.
   * ***NOTE:*** if the element type of the matrix is not `unsigned char` or `float` (e.g. if `image` is not `arma::Mat<unsigned char>` or `arma::fmat`), the matrix will be temporarily converted during resizing; therefore, using `unsigned char` or `float` as the element type is the most efficient.
```

I wanted to get that caveat in there so that users know what to expect.

---

- `ResizeImages(images, info, newWidth, newHeight)`
   * `images` an armadillo matrix contains a set of images, each one is
     represeted in one column representing the image as a flattened vector.

   * `info` will be populated with information from the image in `image`.

   * `newWidth` the new width of the resized image.

   * `newHeight` the new height of the resized image.

   * This function does not return but it modifies the info object and images
     matrix.

   * This function expects all the images at the start to have identical
     dimensions. If the images have different dimension, please use
     `Resize` and loop over all of them using a `for` loop.

> 

Reviewer: I forgot to write this in the review, but we should also add an example of usage for a single image and for multiple images too.  Probably the example usage is the most important part of the documentation.

mlpack::data::Save(outImages, matrix, info);
```

### Resize images

It is possible to resize images in mlpack with the following function:

- `ResizeImages(images, info, newWidth, newHeight)`

>    * `images` an armadillo matrix contains a set of images, each one is
>      represeted in one column representing the image as a flattened vector.
> 
>    * `info` will be populated with information from the image in `image`.
> 
>    * `newWidth` the new width of the resized image.
> 
>    * `newHeight` the new height of the resized image.

Reviewer: ```suggestion
   * `images` is a [column-major matrix](matrices.md) containing a set of
      images; each image is represented as a flattened vector in one column.

   * `info` is a [`data::ImageInfo&`](#dataimageinfo) containing details about
     the images in `images`, and will be modified to contain the new size of the
     images.

   * `newWidth` and `newHeight` (of type `size_t`) are the desired new
     dimensions of the resized images.
```

I tried to make these read a little more like the other documentation; hopefully I did not mess up any intentions you had for the writing.  I also wanted to make sure to clarify the types.

>    * This function does not return but it modifies the info object and images
>      matrix.

Reviewer: ```suggestion
   * This function returns `void` and modifies `info` and `images`.
```
Seems more concise, up to you.  (At the very least backticks around info/images would be good.)

>    * ***NOTE:*** if the element type of the matrix is not `unsigned char` or
>    `float` (e.g. if `image` is not `arma::Mat<unsigned char>` or `arma::fmat`),
>    the matrix will be temporarily converted during resizing; therefore, using
>    `unsigned char` or `float` as the element type is the most efficient.

Reviewer: ```suggestion
   * ***NOTE:*** if the element type of `images` is not `unsigned char` or
     `float` (e.g. if `image` is not `arma::Mat<unsigned char>` or
     `arma::fmat`), the matrix will be temporarily converted during resizing;
     therefore, using `unsigned char` or `float` as the element type is the most
     efficient.
```

>    * This function expects all the images at the start to have identical
>      dimensions. If the images have different dimension, please assign each
>      image to one armadillo column and loop over all of them using a `for`
>      loop.

Reviewer: ```suggestion
   * This function expects all the images to have identical
     dimensions. If this is not the case, iteratively call `ResizeImages()` with
     a single image/column in `images`.
```

My goal was to make it shorter; you decide if I did a good job. :)

> Example usage of the ResizeImages function on a set of images:

Reviewer: ```suggestion
Example usage of the `ResizeImages()` function on a set of images:
```

> ```c++

Reviewer: I'm not sure if the documentation snippets job will build right for this example.  I know that all the other example code doesn't use the `mlpack` namespace so that it can be trivially pasted, so at least it will probably need a bunch of `mlpack::`s in it.  You can test the compilation locally with this command:

```
CXX=g++ scripts/test-docs.sh doc/user/load_save.md
```

from the root of the repository.

It's awesome that you have both usages (single image and multiple images) in here, but maybe we should have them as separate examples?  I just mean breaking the code block after "Now all the images have the same dimension", adding a little line that says it is an example about resizing images in batch, and then a new code block.
